compiler/rustc_trait_selection/src/traits/select/candidate_assembly.rs

```diff
@@ -171,15 +187,26 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
             let mut distinct_normalized_bounds = FxHashSet::default();
             let _ = self.for_each_item_bound::<!>(
                 placeholder_trait_predicate.self_ty(),
-                |selcx, bound, idx| {
-                    let Some(bound) = bound.as_trait_clause() else {
+                |selcx, clause, idx| {
+                    let Some(bound) = clause.as_trait_clause() else {
                         return ControlFlow::Continue(());
                     };
                     if bound.polarity() != placeholder_trait_predicate.polarity {
                         return ControlFlow::Continue(());
                     }
 
                     selcx.infcx.probe(|_| {
+                        if util::unelaborated_sizedness_candidate(selcx.infcx, obligation, [bound])
+                            .is_some()
+                        {
+                            // As `ProjectionCandidate` takes an index and not a predicate, a
+                            // "corrected" `<Projection>: MetaSized` candidate cannot be created,
+                            // so instead keep the index of the `<Projection>: Sized` predicate and
+                            // correct for this in confirmation.
+                            candidates.vec.push(ProjectionCandidate(idx));
+                            return;
+                        }
+
                         // We checked the polarity already
                         match selcx.match_normalize_trait_ref(
```

Comment:
even for unelaborated `Sized` bounds, check `match_normalize_trait_ref` using the trait ref returned by `unelaborated_sizedness_candidate`

Reply:
Fixed